src/js/jsx/shared/Datalist.jsx

```diff
@@ -519,20 +552,35 @@ define(function (require, exports, module) {
                 filter = this.state.filter,
                 title = this.state.active && filter !== null ? filter : value,
                 searchableFilter = filter ? filter.toLowerCase() : "",
-                filteredOptions = this._filterOptions(searchableFilter, false),
+                filteredOptions = this._filterOptions(searchableFilter, true),
                 searchableOptions = filteredOptions;
 
             if (filteredOptions && collection.uniformValue(collection.pluck(filteredOptions, "type"))) {
                 searchableOptions = this.props.placeholderOption;
             }
             
-            var autocomplete,
-                hiddenTI;
+            // Use hidden text input and hidden svg to find position of suggestion by updating their values and 
+            // getting width before calling render. Without using the hidden elements, in certain cases the 
+            // calculated position of suggestion would be one keystroke behind. 
+
+            // Hidden text input gets the value of the search input before rendering.
+            var hiddenTI,
```

Comment:
I believe React prefers components to be defined, or `null`. Can you null initialize them here?